--- a/systemUi.js
+++ b/systemUi.js
@@ -21,6 +21,7 @@
 var rollingScreenViewState;
 var supportScreenType;
 var AppOnBackKeyPress;
+var homeAppId;
 // READ ONLY:Should change to CONST
 var UNDEFINED_APP_ID = "UNDEFINED_APP_ID";

@@ -42,6 +43,18 @@
                        JSON.stringify({"serviceName":"com.webos.service.rollingscreen", "subscribe":true}),
                        rollingScreenServerStatusChanged, true);

+    checkIsAltHomeInstalled();
+}
+
+function checkIsAltHomeInstalled() {
+    var xhr = new XMLHttpRequest();
+
+    xhr.onreadystatechange = function () {
+        homeAppId = xhr.status == 200 ? 'com.kitsuned.althome' : 'com.webos.app.home';
+    };
+
+    xhr.open('GET', 'file:///media/developer/apps/usr/palm/applications/com.kitsuned.althome/appinfo.json');
+    xhr.send();
 }

 function cb_getConfigs(payload) {
@@ -84,7 +97,7 @@
 }

 function launchHomeApp(key) {
-    if (!access.ribbon || !overlayView.access.ribbon || settingsService.disableHome ||
+    if (!access.ribbon || !overlayView.access.ribbon || settingsService.disableHome || !homeAppId ||
         (bootManager.bootFirstUse && !systemPropertyService.powerOnlyMode)) {
         pmLog.info("HOME_IGNORED", {"access.ribbon": access.ribbon,
                                     "overlayView.access.ribbon" : overlayView.access.ribbon,
@@ -93,7 +106,7 @@
                                     "powerOnlyMode": systemPropertyService.powerOnlyMode}, "", true);
         return;
     }
-    applicationManager.launch("com.webos.app.home",
+    applicationManager.launch(homeAppId,
                               JSON.stringify({"activateType": getActivateType(key)}));
 }

